dace/frontend/operations.py

```diff
@@ -47,34 +57,45 @@ def __init__(self, repetitions: int = 0, warmup: int = 0) -> None:
     def __call__(self, compiled_sdfg: 'CompiledSDFG', args: Tuple[Any, ...]):
         from dace.codegen.instrumentation import report  # Avoid import loop
 
-        start = timer()
+        # zeros to overwrite start time, followed by indices for each repetition
+        iterator = chain(repeat(0, self.warmup), range(1, self.repetitions + 1))
 
-        times = [start] * (self.repetitions + 1)
-        ret = None
-        print('\nProfiling...')
-
-        iterator = range(self.warmup + self.repetitions)
-        if Config.get_bool('profiling_status'):
+        if Config.get_bool("profiling_status"):
             try:
                 from tqdm import tqdm
-                iterator = tqdm(iterator, desc="Profiling", file=sys.stdout)
+
+                iterator = tqdm(
+                    iterator,
+                    desc="Profiling",
+                    total=(self.warmup + self.repetitions),
+                    file=sys.stdout,
+                    leave=self.tqdm_leave,
+                )
             except ImportError:
-                print('WARNING: Cannot show profiling progress, missing optional '
-                      'dependency tqdm...\n\tTo see a live progress bar please install '
-                      'tqdm (`pip install tqdm`)\n\tTo disable this feature (and '
-                      'this warning) set `profiling_status` to false in the dace '
-                      'config (~/.dace.conf).')
+                warnings.warn(
+                    "Cannot show profiling progress, missing optional dependency "
+                    "tqdm...\n\tTo see a live progress bar please install tqdm "
+                    "(`pip install tqdm`)\n\tTo disable this feature (and this "
+                    "warning) set `profiling_status` to false in the dace config "
+                    "(~/.dace.conf)."
```

Comment:
why are the quotes double? see contributing guidelines.

Reply:
I don't think we have it in our contributing guidelines, do we? We just mention that we follow Google's style, and that just states to be consistent; which [we're not](https://github.com/spcl/dace/blob/master/dace/sdfg/sdfg.py#L183) 🙃

Reply:
Unless I'm missing it, we should probably add it

Reply:
oh dear, we should. thx!